core/web/eth_keys_controller.go

```diff
@@ -379,3 +410,25 @@ func (ekc *ETHKeysController) setKeyMaxGasPriceWei(state ethkey.State, keyAddres
 	}
 	return presenters.SetETHKeyMaxGasPriceWei(utils.NewBig(price.ToInt()))
 }
+
+// getChain is a convenience wrapper to retrieve a chain for a given request
+// and call the corresponding API response error function for 400, 404 and 500 results
+func (ekc *ETHKeysController) getChain(c *gin.Context, cs evm.ChainSet, chainIDstr string) (chain evm.Chain, ok bool) {
+	chain, err := getChain(ekc.app.GetChains().EVM, chainIDstr)
+	if err != nil {
+		if errors.Is(err, ErrInvalidChainID) {
+			jsonAPIError(c, http.StatusInternalServerError, err)
+			return nil, false
+		} else if errors.Is(err, ErrMultipleChains) {
+			jsonAPIError(c, http.StatusInternalServerError, err)
+			return nil, false
+		} else if errors.Is(err, ErrMissingChainID) {
+			jsonAPIError(c, http.StatusInternalServerError, err)
+			return nil, false
+		} else {
+			jsonAPIError(c, http.StatusInternalServerError, err)
+			return nil, false
+		}
```

Comment:
Usually the linter insists on these cases being un-indented:
```suggestion
		} 
		jsonAPIError(c, http.StatusInternalServerError, err)
		return nil, false
```